### ISSUE TITLE
UFS-Aerosols component (GOCART) platform / compiler / prod-debug limitations

### ISSUE THREAD

**Issue by climbfuji** [ufs-community/ufs-weather-model#677]
## Description

PR https://github.com/ufs-community/ufs-weather-model/pull/591 adds the UFS-Aeorosols component (GOCART) to the ufs-weather-model. The regression tests are run on Hera with Intel in PROD mode only. Issues/limitations:

1. Hera / Intel / DEBUG regression test fails with
```
  0:  af create fcst fieldbundle, name=sfc_nearest_stodrc=           0
  0:  af create fcst fieldbundle, name=sfc_bilinearrc=           0
  0:  in fcst,init total time:    66.0327100753784
  0:  in fcst run phase 2, na=           0
144:  in readxml,bf readxml,size(post_avblflds%param)=           0
144:  xml_perl_data read Post flat file
144:  allocate paramset to :           2
144:  allocate param to :          45
144:  sum num_post_afld :          45
144:  allocate param to :         149
144:  sum num_post_afld :         194
144:  in readxml, aft read flat file.xml,num_post_afld=         149
144:  in readxml, aft read flat file.xml,num_pset=           2
144:  Size of pset is:          149
144:  datset is: GFSPRS
144:  MXFLD is:          999
144:  size of lvlsxml:        97000
144:  size of post_avblflds param         149
144:  in readxml. nfld=         149 nrecout=         994
144:  post_fname=GFSPRS.GrbF00
144: forrtl: error (65): floating invalid
144: Image              PC                Routine            Line        Source
144: fv3.exe            000000000969D7DE  Unknown               Unknown  Unknown
144: libpthread-2.17.s  00002B96987D2630  Unknown               Unknown  Unknown
144: fv3.exe            0000000009290762  bndlyr_                   311  BNDLYR.f
144: libiomp5.so        00002B9696EE2A43  __kmp_invoke_micr     Unknown  Unknown
144: libiomp5.so        00002B9696EA62C6  __kmp_fork_call       Unknown  Unknown
144: libiomp5.so        00002B9696E65BB0  __kmpc_fork_call      Unknown  Unknown
144: fv3.exe            0000000009295D5E  bndlyr_                   272  BNDLYR.f
144: fv3.exe            00000000091FB669  miscln_                  1712  MISCLN.f
144: fv3.exe            0000000008F49327  process_                  116  PROCESS.f
144: fv3.exe            0000000003E9D321  post_gfs_mp_post_         201  post_gfs.F90
144: fv3.exe            0000000003E98BDA  inline_post_mp_in          43  inline_post.F90
144: fv3.exe            0000000003E36DE0  module_wrt_grid_c        1489  module_wrt_grid_comp.F90
144: fv3.exe            000000000132E79B  _ZN5ESMCI6FTable1        2036  ESMCI_FTable.C
144: fv3.exe            000000000132A15C  ESMCI_FTableCallE         765  ESMCI_FTable.C
144: fv3.exe            0000000001FF2A42  _ZN5ESMCI3VMK5ent        2204  ESMCI_VMKernel.C
144: fv3.exe            0000000001F758D0  _ZN5ESMCI2VM5ente        1211  ESMCI_VM.C
144: fv3.exe            000000000132A845  c_esmc_ftablecall         922  ESMCI_FTable.C
144: fv3.exe            00000000010C162C  esmf_compmod_mp_e        1214  ESMF_Comp.F90
144: fv3.exe            00000000017797AD  esmf_gridcompmod_        1889  ESMF_GridComp.F90
144: fv3.exe            0000000003AF9513  fv3gfs_cap_mod_mp        1263  fv3_cap.F90
144: fv3.exe            0000000001E2CB8C  _ZNK5ESMCI13Metho         333  ESMCI_MethodTable.C
144: fv3.exe            0000000001E2DFF0  _ZN5ESMCI11Method         519  ESMCI_MethodTable.C
144: fv3.exe            0000000001E2C9A0  c_esmc_methodtabl         303  ESMCI_MethodTable.C
```

2. `APP=ATMAERO` doesn't compile with GNU 9.2.0 on Hera (not tried: GNU 10.1.0 on Cheyenne):
```
/scratch1/BMC/gmtb/Dom.Heinzeller/ufs-weather-model/ufs-weather-model-feature-gocart/gnu/GOCART/ESMF/UFS/Aerosol_Comp_Mod.F90:38:33:

   38 |   character(len=*), dimension(*,2), parameter :: &
      |                                 1
Error: Bad specification for assumed size array at (1)
/apps/cmake/3.20.1/bin/cmake -E cmake_copy_f90_mod include/UFS_Aerosols/aerosol_diag_mod.mod GOCART/ESMF/UFS/CMakeFiles/UFS_Aerosols.dir/aerosol_diag_mod.mod.stamp GNU
/apps/cmake/3.20.1/bin/cmake -E touch GOCART/ESMF/UFS/CMakeFiles/UFS_Aerosols.dir/Aerosol_Diag_Mod.F90.o.provides.build
/scratch1/BMC/gmtb/Dom.Heinzeller/ufs-weather-model/ufs-weather-model-feature-gocart/gnu/GOCART/ESMF/UFS/Aerosol_Comp_Mod.F90:201:34:

  201 |       fieldMapSize = size(fieldMap, 1)
      |                                  1
```
3. `APP=ATMAERO` not tested on other R&D tie 1 platforms (orion, jet, gaea, cheyenne) / with other compilers due to issues with compiler versions, dependencies, ...

**Comment by rmontuoro** [comment]
I've submitted a PR to address the build issue with GNU 9.2.0 for the UFS Aerosols component.

**Comment by rmontuoro** [comment]
Commit [7d59f3](https://github.com/ufs-community/ufs-weather-model/pull/591/commits/7d59f31c65fd830544ad619d9df0625fea8b7d5c) fixes the build issue with GNU 9.2.0 on Hera.

**Comment by junwang-noaa** [comment]
Add several other issues:

4. What software are required and if they can be installed on operational platforms wcoss2/dell (miniconda/panda dependencies, etc) so that the model can be tested in operational platforms?

5. How to resolve the dependencies of python/panda, yafyaml dependencies brought in by this PR for long term? Who is going to maintain those libraries on all the platforms?

**Comment by arunchawla-NOAA** [comment]
Can we list all the dependencies so we can review with NASA on the path forward ?

**Comment by arunchawla-NOAA** [comment]
@rmontuoro and @junwang-noaa is this still an issue for gocart build ?

**Comment by rmontuoro** [comment]
@arunchawla-NOAA - I believe all issues mentioned here have been addressed:
- The GNU 9.2.0 build issue was addressed in PR #591 by commit 7d59f31c65fd830544ad619d9df0625fea8b7d5c
- Python/panda is no longer required. Current dependencies are: https://github.com/ufs-community/ufs-weather-model/blob/38aa6340894bd6320fde87ffe1dcee3c0d393c08/modulefiles/ufs_common#L21-L23
- UFS-Aerosols runs on level 1 platforms. See, for instance, PR #1071.

@junwang-noaa - please confirm.

**Comment by junwang-noaa** [comment]
That is correct. We can close this issue when PR#1071 is committed.
